components/ee/ws-scheduler/pkg/scheduler/scheduler.go

```diff
@@ -290,22 +305,51 @@ func (s *Scheduler) schedulePod(ctx context.Context, pod *corev1.Pod) (err error
 		return nil
 	}
 
-	// we found a node for this pod - bind it to the node
-	err = s.bindPodToNode(ctx, pod, nodeName)
-	if err != nil {
-		errStr := err.Error()
-		if strings.Contains(errStr, "is already assigned to node") {
-			// some other worker was faster: This can happen and is good: do nothing
-			log.WithFields(owi).WithField("name", pod.Name).Debugf("pod already bound - someone was faster than me!")
-			return nil
+	// mark as already scheduled, even before the actually scheduling has happened.
+	// this enables asynchronous scheduling - we just need to make sure we release the binding in case the scheduling fails!
+	s.localBindingCache.markAsScheduled(pod, nodeName)
+
+	// we found a node for this pod: (asynchronously) bind it to the node! schedulingPodMap makes sure we do not try to
+	// select the same slot multiple times.
+	go func() {
+		var err error
+		defer s.schedulingPodMap.remove(pod.Name)
+		defer func() {
+			if err != nil {
+				// make sure we already release the binding - but only:
+				//  - in the error case
+				//  - when we _know_ the pod is already assigned (k8s error "already assigned to node")
+				s.localBindingCache.delete(pod.Name)
+			}
+		}()
+
+		// if this is a regular workspace: try to find a ghost and delete it
+		if isRegularWorkspace {
+			ghostToDelete := state.FindOldestGhostOnNode(nodeName)
```

Comment:
Potential double selection of `ghostToDelete`

Reply:
Fixed with https://github.com/gitpod-io/gitpod/pull/2552/commits/beec4278847a324f783d6257ae083362dfad6277.